### PR TITLE
STRF-6701 Reduce usage of Lodash

### PR DESCRIPTION
**Pull request by bc-williamkwon**
Storefront-6701 reduce usage lodash in order to improve performance.  The test cases pass with this refactor. @mattolson

### CODE REVIEW

File: lib/translator/locale-parser.js [hunk 1]
```diff
@@ -18,14 +17,12 @@ function getPreferredLocale(acceptLanguage, languages, defaultLocale) {
...
-    _.each(locales, locale => {
-        if (languages[locale]) {
-            preferredLocale = locale;
-
-            return false;
+    for (let i = 0; i < locales.length; i++) {
+        if (languages[locales[i]]) {
+            preferredLocale = locales[i];
+            break;
         }
```
Comment (bc-zoharmuzafi) [inline lib/translator/locale-parser.js]: maybe you can use Array.foreach just a suggestion

**Comment by bc-zoharmuzafi** [comment]
@mattolson should we adjust 2x as well?

File: lib/translator/locale-parser.js (hunk 1 above)
Comment (bc-williamkwon) [reply lib/translator/locale-parser.js]: Issue with forEach is that it doesn't have the ability to break.

File: lib/translator/filter.js [hunk 2]
```diff
@@ -4,24 +4,30 @@
...
+            filteredTranslations[languageKey] = language[languageKey];
         }
     });
+    return filteredTranslations;
 }
 
 module.exports = {
     filterByKey: filterByKey,
-};
+};
```
[the remark this answers is not in the document]
Comment (bc-williamkwon) [reply lib/translator/filter.js]: fixed.

**Pull request closed without merging**